src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -0,0 +1,210 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ConflictMarkerResolution
+{
+    internal abstract class AbstractResolveConflictMarkerCodeFixProvider : CodeFixProvider
+    {
+        protected AbstractResolveConflictMarkerCodeFixProvider(string diagnosticId)
+        {
+            FixableDiagnosticIds = ImmutableArray.Create(diagnosticId);
+        }
+
+        protected abstract bool IsNewLine(char ch);
+        protected abstract bool IsEndOfLine(SyntaxTrivia trivia);
+        protected abstract bool IsDisabledText(SyntaxTrivia trivia);
+        protected abstract bool IsConflictMarker(SyntaxTrivia trivia);
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; }
+
+        private static readonly int s_mergeConflictLength = "<<<<<<<".Length;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+            var startTrivia = root.FindTrivia(context.Span.Start);
+
+            if (!IsConflictMarker(text, startTrivia, '<'))
+            {
+                return;
+            }
+
+            var token = startTrivia.Token;
+
+            while (true)
+            {
+                token = token.GetNextToken(includeZeroWidth: true);
+                if (token.RawKind == 0)
+                {
+                    return;
+                }
+
+                var index = GetEqualsConflictMarkerIndex(text, token);
+                if (index >= 0)
```

Comment:
Verifying my understanding of the compiler side: the <<<<<<< is only treated as a conflict marker if there exists a ======= and a >>>>>>> to match, right? Otherwise this index would always be -1 and you'd be stuck in a loop either forever or until GetNextToken fails, I'm not sure which.